src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -56,45 +54,57 @@ def solve(self, firststep, nsteps=1, **kwargs):
 
         ####### This function surely can use some code cleanup. #######
 
-        current = self.gfilt.initialdistribution
+        current_rv = self.gfilt.initialrandomvariable
         step = firststep
-        ssqest, ct = 0.0, 0
-        times, means, covars = [self.ivp.t0], [current.mean()], [current.cov()]
+        ssqest, n_steps = 0.0, 0
+        times, means, covars = [self.ivp.t0], [current_rv.mean()], [current_rv.cov()]
         while times[-1] < self.ivp.tmax:
             intermediate_step = float(step / nsteps)
-            tm = times[-1]
+            t = times[-1]
+
+            pred_rv = current_rv
             interms, intercs, interts = [], [], []
             for idx in range(nsteps):
-                newtm = tm + intermediate_step
-                current, __ = self.gfilt.predict(tm, newtm, current, **kwargs)
-                interms.append(current.mean().copy())
-                intercs.append(current.cov().copy())
-                interts.append(newtm)
-                tm = newtm
-            predicted = current
-            new_time = tm
+                t_new = t + intermediate_step
+                pred_rv, __ = self.gfilt.predict(t, t_new, pred_rv, **kwargs)
+                interms.append(pred_rv.mean().copy())
+                intercs.append(pred_rv.cov().copy())
+                interts.append(t_new)
+                t = t_new
+
+            t_new = t
```

Comment:
this is just intermediate, no? If I understand correctly, this block will be gone with the next PR anyway?

Reply:
Yep, this is gone with the dense-output stuff. But I still use `t` and `t_new`, and `pred_rv` as variable names there.

Reply:
Fair enough. I don't see any problem here (for now), was just wondering.